pkg/daemon/update.go

```diff
@@ -625,7 +622,16 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) (retErr err
 		return err
 	}
 
-	return dn.performPostConfigChangeAction(actions, newConfig.GetName())
+	if err := dn.performPostConfigChangeAction(actions, newConfig.GetName()); err != nil {
+		return err
+	}
+
+	// if dn.skipReboot, a reboot might be performed manually after update(), in which case we don't want to update state
+	// else reboot was not needed to apply the configuration, and we need to update state
+	if !dn.skipReboot {
```

Comment:
I'm a bit confused at the use of skipReboot here. This is only for onceFrom right? Why is this specifically handled here?

Reply:
I missed adding this condition originally and it was causing bootstrap to fail. Looking at the old version of `performPostConfigChangeAction` it calls `dn.reboot` which I initially assumed would never return. But if `dn.skipReboot == true`, `dn.reboot` does return, and `performPostConfigChangeAction` returns early before updating state. So I had to add this to preserve that behavior. I can't say I fully understand why that's necessary because how state is managed is pretty unclear to me